compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -417,12 +406,42 @@ impl<'context> Elaborator<'context> {
         meta.function_body = FunctionBody::Resolved;
 
         self.trait_bounds.clear();
-        self.type_variables.clear();
         self.interner.update_fn(id, hir_func);
         self.current_function = old_function;
         self.current_item = old_item;
     }
 
+    /// Defaults all type variables used in this function context then solves
+    /// all still-unsolved trait constraints in this context.
+    fn check_and_pop_function_context(&mut self) {
```

Comment:
Nitpick: should this be `pop_and_check_function_context`?

Reply:
Maybe? These read very similarly to me. I think I'll leave the name for now